### ISSUE TITLE
Crashes everywhere

### ISSUE THREAD

**Issue by martinrotter** [moneymanagerex/android-money-manager-ex#1204]
1. I enable sync, then I select Dropbox -> app crashes.
2. I open sync again, Dropbox is already selected, I select "Remote file", enter credentials, then I select remote file -> app crashes.
3. I open sync again, Remote file is selected, I hit "Download file" -> app crashes.
4. At this point, app cannot be started anymore, crashes right on startup.

REALLY?

I mean, I am quite calm person, source contributor to multiple project. I even considered donating/providing patches, but these so crazy multiple errors made me so angry, that I completely uninstall even desktop version, because at this point, whole solution is just unusable.

**Comment by MisterY** [comment]
Perhaps you could help identify the cause of the issue on your device? It works just fine for thousands of active daily users. I haven't had a problem with Dropbox sync in years and I have been using the app almost daily.

**Comment by MisterY** [comment]
Perhaps you could help identify the cause of the issue on your device? It works just fine for thousands of active daily users. I haven't had a problem with Dropbox sync in years and I have been using the app almost daily.

**Comment by MisterY** [comment]
Perhaps you could help identify the cause of the issue on your device? It works just fine for thousands of active daily users. I haven't had a problem with Dropbox sync in years and I have been using the app almost daily.

**Comment by MisterY** [comment]
Perhaps you could help identify the cause of the issue on your device? It works just fine for thousands of active daily users. I haven't had a problem with Dropbox sync in years and I have been using the app almost daily.

**Comment by MisterY** [comment]
Perhaps you could help identify the cause of the issue on your device? It works just fine for thousands of active daily users. I haven't had a problem with Dropbox sync in years and I have been using the app almost daily.

**Comment by MisterY** [comment]
Perhaps you could help identify the cause of the issue on your device? It works just fine for thousands of active daily users. I haven't had a problem with Dropbox sync in years and I have been using the app almost daily.

**Comment by MisterY** [comment]
Perhaps you could help identify the cause of the issue on your device? It works just fine for thousands of active daily users. I haven't had a problem with Dropbox sync in years and I have been using the app almost daily.

**Comment by MisterY** [comment]
Perhaps you could help identify the cause of the issue on your device? It works just fine for thousands of active daily users. I haven't had a problem with Dropbox sync in years and I have been using the app almost daily.

**Comment by MisterY** [comment]
Perhaps you could help identify the cause of the issue on your device? It works just fine for thousands of active daily users. I haven't had a problem with Dropbox sync in years and I have been using the app almost daily.

**Comment by MisterY** [comment]
Perhaps you could help identify the cause of the issue on your device? It works just fine for thousands of active daily users. I haven't had a problem with Dropbox sync in years and I have been using the app almost daily.

**Comment by velmuruganc** [comment]
Sync with Dropbox works perfectly works fine for me. I didnt find any issue(s). I am using Sony Z5.

**Comment by martinrotter** [comment]
Well, I can install it again. How to produce any logs? What exactly should I do?

**Comment by MisterY** [comment]
If you are online, all the exception logs should be delivered automatically by CrashLytics once you restart the app. You can also try the Beta version from the PlayStore.
There are no currently no other logging options as it was too difficult to maintain.

**Comment by martinrotter** [comment]
I tried Beta, behaves same as stable.

**Comment by martinrotter** [comment]
In that case, your exception logs should be already delivered because I was connected to wifi when bugs happened.

**Comment by MisterY** [comment]
Definitely something with that device or the CloudRail library.

**Comment by martinrotter** [comment]
Which device? My phone? Pretty standard Xiaomi Redmi bro, nothing fancy.

**Comment by MisterY** [comment]
Redmi 3 or 4?

**Comment by MisterY** [comment]
This could be you:
![image](https://user-images.githubusercontent.com/462445/36305335-6aeae79e-1313-11e8-9788-fcf0f374c01f.png)

**Comment by martinrotter** [comment]
Redmi 2. I was testing it yesterday afternoon (central euro time).

**Comment by MisterY** [comment]
It seems that many Redmi devices crash with this:
```
Fatal Exception: java.lang.NoSuchMethodError: No static method isCreatable(Ljava/lang/String;)Z in class Lorg/apache/commons/lang3/math/NumberUtils; or its super classes (declaration of 'org.apache.commons.lang3.math.NumberUtils' appears in /system/framework/framework.jar:classes2.dex)
       at com.money.manager.ex.settings.SyncPreferences.getSyncInterval(SyncPreferences.java:65)
       at com.money.manager.ex.sync.SyncSchedulerBroadcastReceiver.startHeartbeat(SyncSchedulerBroadcastReceiver.java:93)
       at com.money.manager.ex.sync.SyncSchedulerBroadcastReceiver.onReceive(SyncSchedulerBroadcastReceiver.java:80)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:2732)
       at android.app.ActivityThread.access$1800(ActivityThread.java:153)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1428)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5441)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:738)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:628)
```

**Comment by martinrotter** [comment]
Android 4.4.4 + MIUI 8.1.1

**Comment by MisterY** [comment]
And, to be honest, I have no clue what to do about it.

**Comment by martinrotter** [comment]
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/math/NumberUtils.html#isCreatable-java.lang.String-

It that library apache commons 3 referenced by the app? Maybe you could check if you have correct version there. Unfortunately, I am not JAVA developer, so maybe this does not make sense. It seems you try to call method, which is not available in apache commons 3 library which is on my phone?

**Comment by martinrotter** [comment]
https://stackoverflow.com/questions/32835143/how-to-add-org-apache-commons-lang3-to-androidstudio-with-gradle/32835663

**Comment by MisterY** [comment]
Yes, that's most likely the reason. The problem here is that this works just fine for more than 10K users. If you wish, you can fix this for your specific device and submit a pull request that does not break compatibility with everyone else and a zillion other devices out there. I'll be happy to merge that and publish a new Beta version for testing.

**Comment by MisterY** [comment]
#1250

**Comment by stefandriy** [comment]
Just faced the same issue. Simple debugging led to the following hot-fix: #1261

**Comment by MisterY** [comment]
Thanks a lot, @stefandriy, for the simple and effective fix! I can't really test this but hopefully other Xiaomi users can.
